src/node_file.cc

```
@@ -2727,6 +2730,238 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
+      return;
+
+    FromNamespacedPath(&initial_file_path);
+
+    for (int i = 0; i < legacy_main_extensions_with_main_end; i++) {
+      file_path = initial_file_path + legacy_main_extensions[i];
+
+      switch (FilePathIsFile(env, file_path)) {
+        case FilePathIsFileReturnType::kIsFile:
+          return args.GetReturnValue().Set(i);
+        case FilePathIsFileReturnType::kIsNotFile:
+          continue;
+        case FilePathIsFileReturnType::kThrowInsufficientPermissions:
+          // the default behavior when do not have permission is to return
+          // and exit the execution of the method as soon as possible
+          // the internal function will throw the exception
+          return;
+        default:
+          UNREACHABLE();
+      }
+    }
+  }
+
+  file_path_url = ada::parse<ada::url_aggregator>("./index", &base.value());
+
+  if (!file_path_url) {
+    return args.GetReturnValue().Set(LegacyMainResolveReturnType::kInvalidUrl);
+  }
+
+  if (!FileURLToPathImpl(env, file_path_url.value(), initial_file_path)) return;
+
+  FromNamespacedPath(&initial_file_path);
+
+  for (int i = legacy_main_extensions_with_main_end;
+       i < legacy_main_extensions_package_fallback_end;
+       i++) {
+    file_path = initial_file_path + legacy_main_extensions[i];
+
+    switch (FilePathIsFile(env, file_path)) {
+      case FilePathIsFileReturnType::kIsFile:
+        return args.GetReturnValue().Set(i);
+      case FilePathIsFileReturnType::kIsNotFile:
+        continue;
+      case FilePathIsFileReturnType::kThrowInsufficientPermissions:
+        // the default behavior when do not have permission is to return
+        // and exit the execution of the method as soon as possible
+        // the internal function will throw the exception
+        return;
+      default:
+        UNREACHABLE();
+    }
+  }
+
+  return args.GetReturnValue().Set(
+      LegacyMainResolveReturnType::kModuleNotFound);
```

Can you throw an error in here, instead of returning `kModuleNotFound`?

---

Throw error on C++ side is faster than JS side?

---

It's not faster, but we don't need to make it faster, since this will stop the execution. So the question is `Do we need to crash faster`?